core/lsp/PreemptionTaskManager.cc

```diff
@@ -32,7 +33,7 @@ bool PreemptionTaskManager::trySchedulePreemptionTask(std::shared_ptr<Task> task
     return success;
 }
 
-bool PreemptionTaskManager::tryRunScheduledPreemptionTask() {
+bool PreemptionTaskManager::tryRunScheduledPreemptionTask(core::GlobalState &gs) {
```

Comment:
During my previous preemption PR, I didn't account for the fact that the error queue needs to be replaced during preemption!

The error queue is where typechecking puts all typechecking errors. For a given edit, Sorbet LSP runs typechecking and then drains the error queue. If we failed to temporarily swap it out during preemption, the preempted task will see all of the errors that have accumulated thus far on the slow path.

I'm not 100% happy with this solution. I debated if it would be cleaner to replace the error queue internally during every typechecking operation to ensure that every task has a fresh queue.

Reply:
Let's leave this comment in source?

Reply:
Sure, can do.

Reply:
(Done)